imagetest/testworkflow.go

```diff
@@ -99,7 +99,75 @@ func (t *TestWorkflow) CreateTestVM(name string) (*TestVM, error) {
 	return &TestVM{name: vmname, testWorkflow: t}, nil
 }
 
-func (t *TestWorkflow) appendCreateVMStep(name, hostname string) (*daisy.Step, error) {
+// CreateTestVMWithCustomNetwork creates the necessary steps to create a VM with  the specified name and custom network to the workflow.
+func (t *TestWorkflow) CreateTestVMWithCustomNetwork(name, networkName,
```

Comment:
the model we have used so far is to call createvm and then call methods on the returned vm object that modify it, for example to start it with secure boot enabled. i don't think we want a pattern of many custom-vm-type creation methods